File: 8_count_sunday.py
start_day = input()
n = int(input())

days = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"]

start = days.index(start_day)

count = 0

for i in range(n):
    current_day = (start + i) % 7

    if current_day == 0:
        count += 1

print(count)

File: 8_count_sunday.cpp
#include<iostream>
#include<unordered_map>
using namespace std;
int main(){
    string day;
    int n;
    cin>>day;
    cin>>n;
    unordered_map<string, int> mp={
        {"sun",0},
        {"mon",1},
        {"tue",2},
        {"wed",3},
        {"thur",4},
        {"fri",5},
        {"sat",6}
    };
    int start=mp[day];
    int daystosunday=(7-start)%7;
    if(daystosunday>=n){
        cout<<0;
    }else{
        cout<<1+(n-daystosunday)/7;
    }
    return 0;
}